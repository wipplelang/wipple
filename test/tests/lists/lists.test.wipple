-- [should compile]

my-list : 1 , 2 , 3

show (my-list . first)
show (my-list . last)
show ((my-list . initial) or unreachable . last)
show ((my-list . tail) or unreachable . first)
show (my-list . nth 1)
