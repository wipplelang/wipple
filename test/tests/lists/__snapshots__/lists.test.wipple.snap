// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/Users/wilson/Desktop/wipple-wasm/test/tests/lists/lists.test.wipple 1`] = `[]`;

exports[`/Users/wilson/Desktop/wipple-wasm/test/tests/lists/lists.test.wipple 2`] = `
{
  "output": "Some 1
Some 3
1: {"type":"jumpIfNot","value":["type Maybe / variant Some",5]}: expected variant
while evaluating {
    "typeDescriptor": {
        "type": "named",
        "value": [
            "type Maybe",
            [
                {
                    "type": "named",
                    "value": [
                        "type List",
                        [
                            {
                                "type": "named",
                                "value": [
                                    "type Number",
                                    []
                                ]
                            }
                        ]
                    ]
                }
            ]
        ]
    },
    "type": "list",
    "values": [
        {
            "typeDescriptor": {
                "type": "named",
                "value": [
                    "type Number",
                    []
                ]
            },
            "type": "number",
            "value": "1"
        },
        {
            "typeDescriptor": {
                "type": "named",
                "value": [
                    "type Number",
                    []
                ]
            },
            "type": "number",
            "value": "2"
        }
    ]
}
stack: [
    {
        "typeDescriptor": {
            "type": "named",
            "value": [
                "type Maybe",
                [
                    {
                        "type": "named",
                        "value": [
                            "type List",
                            [
                                {
                                    "type": "named",
                                    "value": [
                                        "type Number",
                                        []
                                    ]
                                }
                            ]
                        ]
                    }
                ]
            ]
        },
        "type": "list",
        "values": [
            {
                "typeDescriptor": {
                    "type": "named",
                    "value": [
                        "type Number",
                        []
                    ]
                },
                "type": "number",
                "value": "1"
            },
            {
                "typeDescriptor": {
                    "type": "named",
                    "value": [
                        "type Number",
                        []
                    ]
                },
                "type": "number",
                "value": "2"
            }
        ]
    }
]
",
}
`;
